templates/_base.html

```diff
@@ -65,45 +65,18 @@
 
   <script nonce="{{nonce}}">
     function onSignIn(googleUser) {
-      var profile = googleUser.getBasicProfile();
-      var id_token = googleUser.getAuthResponse().id_token;
-        var xhr = new XMLHttpRequest();
-      xhr.open('POST', '/api/v0/login');
-      xhr.setRequestHeader('Content-Type', 'application/json');
-      xhr.onload = function() {
-        if (xhr.status == 200) {
-            if (window.location.href.includes('?')) {
-                window.location.replace(window.location.href.split("?")[0])
-            } else {
-                window.location.reload();
-            }
-        }
-        else {
-          // signout if cookie not set
-          signOut();
-        }
-      };
-      let data = JSON.stringify({ "id_token": id_token})
-      xhr.send(data);
-
+        csClient.signIn(googleUser).then(response => {
+            console.log(response);
```

Comment:
In case of error (i.e. cookie is not set due to some reason in the backend), `signOut()` must be called so that user is logged out using the Google Sign-In library. Otherwise, in case of the error, Google Sign-In would consider user has logged in but our backend would consider the user has not logged in.

Reply:
Good point.  That will help avoid user confusion in the case where something goes wrong.   Added.